include/measurement_kit/http/http.hpp

```diff
@@ -328,7 +327,7 @@ class Stream {
      */
     void on_connect(std::function<void(void)>&& fn) {
         connect_handler = fn;
-        connection->on_connect([this]() {
+        connection.on_connect([this]() {
```

Comment:
The benefit of not using a shared pointer is simpler code (hence simpler review and most importantly simpler debugging). On the contrary the benefit of using a shared pointer is that the pointee is alive as long as the shared pointer is alive. At a certain point this looked promising because it could allow one to emulate the semantic of languages like JavaScript. In particular I explored the possibility of using lambda closures to keep objects alive.

However, after trying this strategy out, I become convinced it was not so good.The code was becoming more complex to read. It was not very reliable to store objects inside the lambda closure (strange implementation issues, leaks because there is no cycle-aware garbage collector, etc.). As an example, see the issues reported in #134 and #111.

So I decided the wisest thing to do was to remove complexity (and probably it would have been better to apply KISS since the beginning). This payed off because I finally understood, for example, the reason of #134 (the fix for which is in a branch that will soon become a pull request).